goss-testing/scripts/check_bgp_neighbors_established.sh

```diff
@@ -49,19 +49,47 @@ function err_exit
     exit $rc
 }
 
-kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' > $TMPFILE ||
-    err_exit 10 "ERROR: Command failed (rc=$?): kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' > $TMPFILE"
+TOKEN=$(curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=`kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d` https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
+
+# check if metalLB configmap exists
+echo "Checking for cofigmap called metallb." > $TMPFILE
+kubectl -n metallb-system get cm metallb 2>&1 |grep -i error|wc -l >> $TMPFILE
+metallb_cm=$(kubectl -n metallb-system get cm metallb 2>&1 |grep -i error|wc -l)
+
+# If error checking for metallb configmap
+if [ "$metallb_cm" -gt "0" ];then
+	metallb_check="0"
+else
+    metallb_check="1"
+fi
+
+# check SLS Networks data for BiCAN toggle
+echo "Checking for BiCAN toggle."  >> $TMPFILE
+curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN|jq -r .ExtraProperties.SystemDefaultRoute | grep -e CHN -e CAN |wc -l >> $TMPFILE
```

Comment:
This doesn't check to see if the curl or jq commands passed or not. I would replace this line and the next line with the following:
```bash
curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN |
    jq -r .ExtraProperties.SystemDefaultRoute |
    tee $TMPFILE ||
        err_exit 30 "Error writing to $TMPFILE or command pipeline failed with return code $?: curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN | jq -r .ExtraProperties.SystemDefaultRoute"
# Set sls_network_check to 0 if the string CHN or CAN appears in the output of the jq command
grep -e CHN -e CAN $TMPFILE
sls_network_check=$?
```

Reply:
i need to check if exists and not stop on error since this is another scenario where this is a specific csm-1.2+ data element.

Reply:
Oh, so you are saying that there are some 1.0 systems where the curl command would succeed to that BICAN endpoint, but we just wouldn't see the CHN or CAN strings in the output from it?

Reply:
Assuming you want to verify both that the BICAN endpoint works and that those strings are there, I would change this section to:
```bash
curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN |
    jq -r .ExtraProperties.SystemDefaultRoute |
    grep -e CHN -e CAN $TMPFILE
# Set sls_network_check to 1 if the BICAN endpoint responds and 
# has the CAN or CHN strings in its ExtraProperties.SystemDefaultRoute field. Otherwise set to 0
[[ $? -eq 0 ]] && sls_network_check=1 || sls_network_check=0
```

Reply:
another improvement!